examples/react-router/app/components/recipe.jsx

```diff
@@ -16,7 +17,7 @@ var Recipe = React.createClass({
   },
 
   getStateFromFlux: function() {
-    var params = this.context.router.getCurrentParams();
+    var params = this.props.params;
```

Comment:
I needed to use `nextProps` (instead of `this.props`) received by `componentWillReceiveProps` to get the latest params.

Here's how:

```javascript
getStateFromFlux: function(newProps) {
  var props = nextProps || this.props;
  var params = props.params;
 // ...
},

componentWillReceiveProps: function(nextProps) {
  this.setState(this.getStateFromFlux(nextProps));
}
```